pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CastTransformFunction.java

```diff
@@ -191,7 +194,19 @@ private long[] transformToTimestampValuesSV(ValueBlock valueBlock) {
       int length = valueBlock.getNumDocs();
       initLongValuesSV(length);
       String[] stringValues = _transformFunction.transformToStringValuesSV(valueBlock);
-      ArrayCopyUtils.copyToTimestamp(stringValues, _longValuesSV, length);
+      RoaringBitmap nullBitmap = _transformFunction.getNullBitmap(valueBlock);
```

Comment:
Should we check if null handling is enabled first?

Reply:
Even if query time null handling is disabled, these default string null values will cause the cast transform to fail. IMO it's the right behavior to convert it to epoch value in that case too. We also do the same thing when null handling is enabled, but there the null bitmap will actually end up being used and the final query result for the `null` rows would be `null`.

Both these cases are illustrated in the new tests I added to `NullQueriesFluentTest` btw.